app/routes/account/history.tsx

```diff
@@ -1,6 +1,163 @@
+        sort: { createdAt: SortOrder.Desc },
+        filter: { active: { eq: false } },
+    };
+
+    const res = await getActiveCustomerOrderList(orderListOptions, { request });
+    if (!res.activeCustomer) {
+        return redirect('/sign-in');
+    }
+    return json({
+        orderList: res.activeCustomer.orders,
+        appliedPaginationLimit: zodResult.data.limit,
+        appliedPaginationPage: zodResult.data.page,
+    });
+}
+
+
 
 export default function AccountHistory() {
+    const { orderList, appliedPaginationLimit, appliedPaginationPage } = useLoaderData<typeof loader>();
+    const submit = useSubmit();
+    const transition = useTransition();
+    const showingOrdersFrom = (appliedPaginationPage - 1) * appliedPaginationLimit + 1;
+    const showingOrdersTo = showingOrdersFrom + orderList.items.length - 1;
+
     return (
-        <div className="p-4">Not available</div>
+        <div className="pt-10 relative">
+            {/* Loading-Overlay */}
+            {transition.state !== "idle" && (<div className="absolute top-0 left-0 w-full h-full z-100 bg-white bg-opacity-75"></div>)}
+
+            {orderList.items.length === 0 && (
+                <div className="py-16 text-3xl text-center italic text-gray-300 select-none flex justify-center items-center">
+                    {orderList.totalItems === 0 ? 'Your future orders will appear here' : 'No more orders, end reached'}
+                </div>
+            )}
+            {/* The actual orders */}
+            {orderList.items?.map(item => (
+                // TODO: CHECK THIS ERROR OUT
+                <OrderHistoryItem key={item.code} order={item} isInitiallyExpanded={true} className="mb-10" />
+            ))}
+
+            {/* Pagination */}
+            <div className="flex flex-row justify-between items-center gap-4">
+                <span className="self-start text-gray-500 text-sm ml-4 lg:ml-6 mt-2">Showing orders {showingOrdersFrom} to {showingOrdersTo} of {orderList.totalItems}</span>
+                <ValidatedForm
+                    className="flex flex-col md:flex-row justify-center items-end md:items-center gap-4 lg:gap-6"
+                    validator={orderPaginationValidator}
+                    method="get"
+                    onChange={e => submit(e.currentTarget, { preventScrollReset: true })}
+                    preventScrollReset
+                >
+
+                    <span className="flex gap-4 items-center">
+                        {transition.state !== "idle" && (<ArrowPathIcon className="animate-spin h-6 w-6 text-gray-500" />)}
+                        <Select
+                            name="limit"
+                            required
+                            noPlaceholder
+                            defaultValue={appliedPaginationLimit}
+                        // disabled={transition.state !== "idle"}
```

Comment:
I'm really picky but this could also be removed as it seems unneccessary.

Reply:
Good point!